src/Service/GdsService.php

```diff
@@ -0,0 +1,372 @@
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+
+
+class GdsService
+{
+
+    /**
+     * The call service.
+     *
+     * @var CallService
+     */
+    private CallService $callService;
+
+    /**
+     * The gateway resource service.
+     *
+     * @var GatewayResourceService
+     */
+    private GatewayResourceService $resourceService;
+
+    /**
+     * The file system handle service.
+     *
+     * @var FileSystemHandleService
+     */
+    private FileSystemHandleService $fshService;
+
+    /**
+     * The BRK service.
+     *
+     * @var BrkService
+     */
+    private BrkService $brkService;
+
+    /**
+     * The cache service.
+     *
+     * @var CacheService
+     */
+    private CacheService $cacheService;
+
+    /**
+     * The entity manager.
+     *
+     * @var EntityManagerInterface
+     */
+    private EntityManagerInterface $entityManager;
+
+    /**
+     * The event dispatcher.
+     *
+     * @var EventDispatcherInterface
+     */
+    private EventDispatcherInterface $eventDispatcher;
+
+
+    /**
+     * The service constructor.
+     *
+     * @param CallService              $callService     The call service.
+     * @param GatewayResourceService   $resourceService The resource service.
+     * @param FileSystemHandleService  $fshService      The file system handle service.
+     * @param BrkService               $brkService      The BRK service.
+     * @param CacheService             $cacheService    The cache service.
+     * @param EntityManagerInterface   $entityManager   The entity manager.
+     * @param EventDispatcherInterface $eventDispatcher The event dispatcher.
+     */
+    public function __construct(
+        CallService $callService,
+        GatewayResourceService $resourceService,
+        FileSystemHandleService $fshService,
+        BrkService $brkService,
+        CacheService $cacheService,
+        EntityManagerInterface $entityManager,
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->resourceService = $resourceService;
+        $this->fshService      = $fshService;
+        $this->callService     = $callService;
+        $this->brkService      = $brkService;
+        $this->cacheService    = $cacheService;
+        $this->entityManager   = $entityManager;
+        $this->eventDispatcher = $eventDispatcher;
+
+    }//end __construct()
+
+
+    /**
+     * Creates the content for a soap message to the GDS download service.
+     *
+     * @param DateTime $lastSynced    The date and time the data has last been synced.
+     * @param array    $configuration Whether or not the call is a test call.
+     *
+     * @return array The resulting request message.
+     */
+    private function createRequestMessage(DateTime $lastSynced, array $configuration): array
+    {
+        $now            = new DateTime();
+        $notYetReported = 'true';
+
+        if ($configuration['test'] === true) {
```

Comment:
'test' ?
could this be something better? if so we probably should rename this. And if not an inline comment might be nice

Reply:
It is a function to configure a test mode, so I don't know about some better name :P

Reply:
ohwww, well that is a very fitting name than xD
Maybe testMode could be a better name however 🤔